File: 27. Remove Element/solution.py
class Solution:
    def removeElement(self, nums, val):
        """
        :type nums: List[int]
        :type val: int
        :rtype: int
        """
        if not nums:
            return 0

        i = 0
        n = len(nums)
        while i < n:
            if nums[i] == val:
                nums[i] = nums[n - 1]
                n-=1
            else:
                i+=1
        return n

File: 27. Remove Element/solution.go
func removeElement(nums []int, val int) int {
	n := len(nums)

	for i := 0; i < n; i++ {
		if nums[i] == val {
			nums[i], nums[n-1] = nums[n-1], nums[i]
			n--
			i--
		}
	}

	return n
}